rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/DiscoverJsfSourceFileRuleProvider.java

```diff
@@ -0,0 +1,44 @@
+package org.jboss.windup.rules.apps.javaee.rules;
+
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.loader.RuleLoaderContext;
+import org.jboss.windup.config.metadata.RuleMetadata;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.config.phase.InitialAnalysisPhase;
+import org.jboss.windup.graph.model.FileLocationModel;
+import org.jboss.windup.graph.service.GraphService;
+import org.jboss.windup.rules.apps.javaee.model.JsfSourceFileModel;
+import org.jboss.windup.rules.files.condition.FileContent;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * Discovers JSF files in the input application.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+@RuleMetadata(phase = InitialAnalysisPhase.class)
+public class DiscoverJsfSourceFileRuleProvider extends AbstractRuleProvider
+{
+
+    @Override
+    public Configuration getConfiguration(RuleLoaderContext context)
+    {
+        return ConfigurationBuilder.begin()
+                .addRule()
+                .when(FileContent.matches("{taglib1}").inFileNamed("{*}.{extension}"))
+                .perform(new AbstractIterationOperation<FileLocationModel>() {
+                    @Override
+                    public void perform(GraphRewrite event, EvaluationContext context, FileLocationModel payload) {
+                        GraphService.addTypeToModel(event.getGraphContext(), payload.getFile(), JsfSourceFileModel.class);
+                    }
+                })
+                .where("taglib1")
+                    .matches("(java\\.sun\\.com/jsf/core)|(java\\.sun\\.com/jsf/html)|(http://java\\.sun\\.com/jsf/facelets)")
+                .where("extension")
+                    .matches("(jsp)|(xhtml)")
```

Comment:
the extension could be even something else even it is minor thing, i have seen it for some rare cases ;)

Reply:
I can see that we have mapped for jsp source file also jspx file extension.

Reply:
Yeah, I thought about that, but it will get a bit more difficult if we have to infer the extension from the configuration somehow.

I agree that we should definitely add jspx, though. I'll add that tomorrow. Thanks!